adapt/intent.py

```diff
@@ -56,29 +98,68 @@ def resolve_one_of(tags, at_least_one):
                 resolution[entity_type].append(tag)
         if len(resolution) == len(possible_resolution):
             return resolution
-
     return None
 
 
 class Intent(object):
     def __init__(self, name, requires, at_least_one, optional):
+        """Create Intent object
+
+        Args:
+            name(str): Name for Intent
+            requires(list): Entities that are required
+            at_least_one(list): One of these Entities are required
+            optional(list): Optional Entities used by the intent
+        """
         self.name = name
         self.requires = requires
         self.at_least_one = at_least_one
         self.optional = optional
 
     def validate(self, tags, confidence):
+        """Using this method removes tags from the result of validate_with_tags
+
+        Returns:
+            intent(intent): Results from validate_with_tags
+        """
         intent, tags = self.validate_with_tags(tags, confidence)
         return intent
 
+    def entities(self):
+        """Used to get the Entities the intent is looking for.
+
+        Returns:
+            list: A list of Entities the intent is looking for. This should never be empty but still could be.
+        """
+        entities = []
+        thelist = self.requires + self.at_least_one + self.optional
```

Comment:
there's no reason to declare this as a separate variable, and thelist is a terrible variable name.